packages/core/upload/server/services/extensions/content-manager/__tests__/entity-manager.test.js

```diff
@@ -0,0 +1,162 @@
+'use strict';
+
+const { signEntityMedia } = require('../entity-manager');
+
+const { getService } = require('../../../../utils');
+
+jest.mock('../../../../utils');
+
+describe('Upload | extensions | entity-manager', () => {
+  const modelUID = 'model';
+  const componentUID = 'component';
+
+  const models = {
+    [modelUID]: {
+      attributes: {
+        media: {
+          type: 'media',
+          multiple: false,
+        },
+        media_repeatable: {
+          type: 'media',
+          multiple: true,
+        },
+        compo_media_repeatable: {
+          type: 'component',
+          repeatable: true,
+          component: componentUID,
+        },
+        compo_media: {
+          type: 'component',
+          component: componentUID,
+        },
+      },
+    },
+    [componentUID]: {
+      attributes: {
+        media_repeatable: {
+          type: 'media',
+          multiple: true,
+        },
+        media: {
+          type: 'media',
+          multiple: false,
+        },
+      },
+    },
+  };
+
+  const media = ['media', 'media_1'].map((entry) => ({
+    formats: {
+      thumbnail: {
+        url: `${entry}_thumb`,
+      },
+      large: {
+        url: `${entry}_large`,
+      },
+      small: {
+        url: `${entry}_small`,
+      },
+      medium: {
+        url: `${entry}_medium`,
+      },
+    },
+    url: `${entry}_url`,
+  }));
+
+  describe('signEntityMedia', () => {
+    beforeAll(() => {
+      global.strapi = {
+        plugins: {
+          upload: {},
+        },
+        getModel: jest.fn((uid) => models[uid]),
+      };
+    });
```

Comment:
Maybe we could avoid importing the `getService` method if we do this:

```suggestion
  describe('signEntityMedia', () => {
    let spySignFileUrls;
    beforeEach(() => {
      spySignFileUrls = jest.fn();
      global.strapi = {
        plugins: {
          upload: {
          	service: () => spySignFileUrls
          },
        },
        getModel: jest.fn((uid) => models[uid]),
      };
    });
```


wdyt?